torch/csrc/jit/python/init.cpp

```diff
@@ -1887,9 +1903,22 @@ void initJITBindings(PyObject* module) {
                 return nullptr;
               }),
           py::call_guard<py::gil_scoped_release>());
-  m.def("_is_alias_of", [](const at::Tensor& self, const at::Tensor& other) {
-    return self.is_alias_of(other);
+  m.def("_is_alias_of", [](const py::object& self, const py::object& other) {
+    return toTypeInferredIValue(self).isAliasOf(toTypeInferredIValue(other));
   });
+  m.def(
+      "_contains_alias_of",
+      [](const py::object& self, const py::object& other) {
+        // the overlaps() functions errors out on certain IValue types.
+        // Thus, this function will return false when it errors out because
+        // only cases where tensors are contained are actually meaningful.
+        try {
+          return toTypeInferredIValue(self).overlaps(
```

Comment:
Can move this try/catch into python? I imagine there could be sources of `TORCH_INTERNAL_ASSERT` and some of those asserts are unexpected and should be reported to the user (but all of them will appear as c10::Error and return false here). If you make this change then please update the comment as well

Reply:
I've deleted this try catch and moved the check into python in the next pr.